src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticWebAssetsIntegrationTest.cs

```diff
@@ -276,7 +284,38 @@ public async Task GenerateStaticWebAssetsManifest_IncrementalBuild_ReusesManifes
 
         public Task InitializeAsync()
         {
-            return PackageTestProjects.PackAsync(Output);
+            try
+            {
+                return PackageTestProjects.PackAsync(Output);
+            }
+            catch (Exception ex)
+            {
+                // We observed pack would "complete" in the past but MSBuild would get stuck.
+                // The generated package looked right, but it caused flakyness on our build infrastructure.
+                // We will try and continue running (even if the process didn't complete) and see if we can
+                // complete the test successfully.
+                Output.WriteLine($"Pack failed or did not complete: '{ex}'.");
+                return Task.CompletedTask;
+            }
+        }
+
+        private async Task RestoreWithRetry()
+        {
+            for (var i = 0; i < 3; i++)
+            {
+                try
+                {
+                    var result = await DotnetMSBuild("Restore");
+                    if (result.ExitCode == 0)
+                    {
+                        break;
+                    }
+                }
+                catch
+                {
+                    // Keep retrying if it fails.
+                }
```

Comment:
Is it fine for the test to continue if restore fails 3x?

Reply:
Yeah, let it just break elsewhere